include/reflection-cpp/reflection.hpp

```diff
@@ -811,6 +811,25 @@ std::string Inspect(std::vector<Object> const& objects)
     }
     return str;
 }
+
+template <typename Object, typename Callback>
+void CollectDifferences(const Object& lhs, const Object& rhs, Callback&& callback)
+{
+    template_for<0, CountMembers<Object>>([&]<auto I>() {
+        if constexpr (std::equality_comparable<MemberTypeOf<I, Object>>)
+        {
+            if (GetMemberAt<I>(lhs) != GetMemberAt<I>(rhs))
+            {
+                callback(MemberNameOf<I, Object>, GetMemberAt<I>(lhs), GetMemberAt<I>(rhs));
```

Comment:
I wonder if it might make sense to either pass `I` as template param to the callback, or to simply add an inverse function of `MemberAt<I>(obj)` to get from `(field, object)` back to the index `I`. :thinking:

Reply:
I think we should not restrict callback to be a templated that takes int as a parameter.
We can implement `IndexOf<Object>("member")` but this might be slow

Reply:
meh, `IndexOf<typename Object, auto PtrToMember>()` sounds more reasonable (and can be compile-time)

Reply:
I added one additional overload for this case : https://github.com/contour-terminal/reflection-cpp/pull/19